aws-athena/ctas/reporting-res_report_summary.sql

```diff
@@ -76,8 +92,19 @@ model_values AS (
     LEFT JOIN town_class AS tc
         ON ap.meta_pin = tc.parid
         AND ap.meta_year = tc.model_join_year
-    WHERE ap.run_id IN (SELECT run_id FROM model.final_model)
-        AND tc.property_group IS NOT NULL
+    INNER JOIN final_model_parsed AS fm
+        ON ap.run_id = fm.run_id
+        AND ap.year = fm.year
+        AND (
+            -- If reassessment year, use different models for different towns
+            (
+                CONTAINS(fm.townships, ap.township_code)
+                AND ap.meta_triad_code = fm.triad_code
+            )
+            -- Otherwise, just use whichever model is "final"
+            OR (ap.meta_triad_code != fm.triad_code AND fm.is_final)
+        )
```

Comment:
This heinous join is the result of the new `final_model` schema. We need a way to get a model per township in a tri year, then the "final" model everywhere else.